### ISSUE TITLE
ArgumentNullExceptions from within SDK when looking at objects in HoloLens

### ISSUE THREAD

**Issue by CookieHael** [microsoft/MixedRealityToolkit-Unity#5494]
## Describe the bug

When in a HoloLens application, the app just randomly crashes and restarts quickly, throwing an error in the output log. This usually happens just once when looking at an object in the beginning, then doesnt happen anymore. 
EDIT: Seems to happen on first detection of my hand
EDIT2: Ok nope, also happens without hand present
EDIT 3: By crashing i mean that it just throws a big exception, the view is gone for a while and then pops back up. I am interning for a company this summer working on a HoloLens project, but my internship will end soon, so I will probably not look at this again :( 

Error output: 

Exception thrown: 'System.NullReferenceException' in Microsoft.MixedReality.Toolkit.SDK.dll
'UnityTest.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\Users\DefaultAccount\AppData\Local\DevelopmentFiles\Template3DVS.Release_x86.rhelsen\System.Diagnostics.StackTrace.dll'. 
'UnityTest.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\Users\DefaultAccount\AppData\Local\DevelopmentFiles\Template3DVS.Release_x86.rhelsen\System.Reflection.Metadata.dll'. 
'UnityTest.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\Users\DefaultAccount\AppData\Local\DevelopmentFiles\Template3DVS.Release_x86.rhelsen\System.IO.FileSystem.dll'. 
'UnityTest.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\Users\DefaultAccount\AppData\Local\DevelopmentFiles\Template3DVS.Release_x86.rhelsen\System.Collections.Immutable.dll'. 
'UnityTest.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\Users\DefaultAccount\AppData\Local\DevelopmentFiles\Template3DVS.Release_x86.rhelsen\System.IO.FileSystem.Primitives.dll'. 
'UnityTest.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\Users\DefaultAccount\AppData\Local\DevelopmentFiles\Template3DVS.Release_x86.rhelsen\System.Runtime.Handles.dll'. 
'UnityTest.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\Users\DefaultAccount\AppData\Local\DevelopmentFiles\Template3DVS.Release_x86.rhelsen\System.Threading.Overlapped.dll'. 
Exception thrown: 'System.IO.FileLoadException' in System.Private.CoreLib.ni.dll
Exception thrown: 'System.IO.FileLoadException' in System.Private.CoreLib.ni.dll
NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.MixedReality.Toolkit.Input.GGVPointer.OnPreSceneQuery()
   at Microsoft.MixedReality.Toolkit.Input.FocusProvider.UpdatePointer(PointerData pointer)
   at Microsoft.MixedReality.Toolkit.Input.FocusProvider.UpdatePointers()
   at Microsoft.MixedReality.Toolkit.Input.FocusProvider.Update()
   at Microsoft.MixedReality.Toolkit.MixedRealityToolkit.<>c.<UpdateAllServices>b__68_0(IMixedRealityService service)
   at Microsoft.MixedReality.Toolkit.MixedRealityToolkit.ExecuteOnAllServices(Action`1 execute)
   at Microsoft.MixedReality.Toolkit.MixedRealityToolkit.UpdateAllServices()
   at Microsoft.MixedReality.Toolkit.MixedRealityToolkit.Update()
   at Microsoft.MixedReality.Toolkit.MixedRealityToolkit.$Invoke53(Int64 instance, Int64* args)
   at UnityEngine.Internal.$MethodUtility.InvokeMethod(Int64 instance, Int64* args, IntPtr method) 
(Filename: <Unknown> Line: 0)

Creating controller model from platform SDK
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug.bindings.h Line: 45)

Failed to obtain controller model from driver
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug.bindings.h Line: 45)

Failed to create controller model from driver, defaulting to BaseController behavior
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug.bindings.h Line: 45)

Creating controller model from platform SDK
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug.bindings.h Line: 45)

Exception thrown: 'System.ArgumentNullException' in System.Runtime.WindowsRuntime.dll
Exception thrown: 'System.ArgumentNullException' in System.Private.CoreLib.ni.dll
ArgumentNullException: Value cannot be null.
Parameter name: source
   at System.WindowsRuntimeSystemExtensions.AsTask[TResult](IAsyncOperation`1 source, CancellationToken cancellationToken)
   at Microsoft.MixedReality.Toolkit.WindowsMixedReality.Input.WindowsMixedRealityController.<CreateControllerModelFromPlatformSDK>d__11.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.<ThrowAsync>b__6_0(Object state)
   at UnityEngine.UnitySynchronizationContext.WorkRequest.Invoke() 
(Filename: <Unknown> Line: 0)

## To reproduce

1. HoloLens app running
2. Look at something

## Expected behavior

No random ArgumentNullExceptions from within the SDK.

## Your Setup (please complete the following information)

- Unity Version 2018.4.4f1
- MRTK Version v2.0 RC2.1

## Target Platform (please complete the following information)

- HoloLens

**Comment by wiwei** [comment]
@CookieHael, a couple of questions here:

1) Can you see if you're still seeing this on this branch: https://github.com/microsoft/MixedRealityToolkit-Unity/tree/prerelease/2.0.0_stabilization
2) Can you attach your min repro here that hits this?

We've seen this in the past in the context of holographic remoting, but in non-remoting scenarios I haven't been able to hit this - if there's a project you have that reliably repros this that would be awesome to share

**Comment by CookieHael** [comment]
Here's a link to the zipped assets and projectsettings: https://we.tl/t-UIih42bLsB

Unpacking these and building etc should run a little project on hololens that reproduces the error fairly reliably. It seems to be a little random when it gets thrown, and it happens only once per run I think.

**Comment by wiwei** [comment]
Closing out some old issues - this one I don't think we ever got to the bottom of (I think that after the info was shared out I was also on leave for a bit, so I'm not sure that folks got a chance to take a peek).

We've made a lot of changes to the MRTK since this however (i.e. since the RC2.1). If folks are still seeing this definitely open a new issue (min repro again being super helpful for identifying what's up).
